pallets/cash/src/core.rs

```diff
@@ -319,6 +330,57 @@ pub fn apply_chain_event_internal<T: Config>(event: &ChainBlockEvent) -> Result<
                 result.to_vec(),
             ),
         },
+        ChainBlockEvent::Matic(_block_num, eth_event) => match eth_event {
+            ethereum_client::EthereumEvent::Lock {
```

Comment:
maybe these should be refactored to `EthLikeEvent` ?

Reply:
Yeah the same thing could probably be said for `ethereum_client` and all of the other things in there EthereumBlock etc.. not sure if that makes the situation clearer or not though